folding-schemes/src/folding/nova/cyclefold.rs

```diff
@@ -193,29 +185,32 @@ where
     pub fn fold_committed_instance(
         // assumes that r_bits is equal to r_nonnat just that in a different format
         r_bits: Vec<Boolean<CF2<C>>>,
-        r_nonnat: NonNativeFieldVar<C::ScalarField, CF2<C>>,
+        r_nonnat: NonNativeUintVar<CF2<C>>,
         cmT: GC,
+        // ci1 is assumed to be always with cmE=0, u=1 (checks done previous to this method)
         ci1: CycleFoldCommittedInstanceVar<C, GC>,
-        // ci2 is assumed to be always with cmE=0, u=1 (checks done previous to this method)
         ci2: CycleFoldCommittedInstanceVar<C, GC>,
```

Comment:
I think that this undoes your [previous PR](https://github.com/privacy-scaling-explorations/sonobe/commit/4dcb981dd4d2644d3671c4e81080b87e750caa79), where the old-wrong comment [was fixed](https://github.com/privacy-scaling-explorations/sonobe/pull/86#discussion_r1561221815)

Reply:
Oops! It seems that I made some mistakes when doing rebase. Will fix this!

Reply:
Right, I think that the branch of this PR contains commits from your previous 2 PRs (the first 8 commits in this PR branch), up until the commit with the message `Reduce the number of constraints in AugmentedFCircuit`, it might be that the repeated commits are an old version from your other PRs and that could be what's causing the undoing of the updates that you did after reviews in the other PRs.

Reply:
Indeed! Now both this and the below one should be fixed.